src/libtest/lib.rs

```diff
@@ -1869,7 +1887,21 @@ mod tests {
             Some(Ok(o)) => o,
             _ => panic!("Malformed arg in parse_ignored_flag"),
         };
-        assert!((opts.run_ignored));
+        assert!((opts.run_ignored == RunIgnored::Only));
+    }
+
+    #[test]
+    fn parse_all_flag() {
+        let args = vec![
+            "progname".to_string(),
+            "filter".to_string(),
+            "--all".to_string(),
+        ];
+        let opts = match parse_opts(&args) {
+            Some(Ok(o)) => o,
+            _ => panic!("Malformed arg in parse_ignored_flag"),
```

Comment:
No need for this panic, we can probably just unwrap?

Reply:
Yeah. Also, I think this is an error message for a buggy test, not a buggy parse_opts.